lib/internal/source_map/source_map_cache.js

```diff
@@ -166,12 +167,16 @@ function sourceMapFromDataUrl(basePath, url) {
 // the sources are resolved relative to the SourceMap (like resolving script
 // src in a html document).
 function sourcesToAbsolute(base, data) {
+  const baseUrl = pathToFileURL(base) + '/';
   data.sources = data.sources.map((source) => {
-    source = (data.sourceRoot || '') + source;
-    if (!/^[\\/]/.test(source[0])) {
-      source = resolve(base, source);
-    }
-    if (!source.startsWith('file://')) source = `file://${source}`;
+    // Instead of manually generating the URL, generate on WHATWG rules
+    try {
+      // If source is any form of full URL (for any protocol)
+      // it will remain full URL. If it is relative, it will
+      // become file:// url, because the baseUrl is file URL.
+      source = (new URL(source, baseUrl)).href;
```

Comment:
I just ended up undertaking close to this same refactor in https://github.com/nodejs/node/pull/35903

Reply:
Yes, that function is affected in both issues. I traced its usage to ensure the first parameter is always a directory path and then converted it to URL. It seems your solution is to first convert it to URL and then call ``sourcesToAbsolute``. Please, make sure it ends with slash ``/``. WHATWG rules for URL will threat the last part as as file otherwise and return one directory upwards. Otherwise both solutions are identical in their core: ``new URL(source, baseURL).href``.

P.S. I am not familiar with the nodejs source, so I try to minimize the amount of dependent changes.